### PR TITLE
32: Dependency cleanup

### LINKED ISSUE #32: Can the version pinning of `pytest` be relaxed?
`pytest` is currently pinned to `<6` in the [setup.py](https://github.com/SasView/sasdata/blob/master/setup.py#L85).

Can I ask what is the reason for this, and can this be relaxed? This is not playing well inside a `conda` environment with `conda` and `pip` packages, as `conda` wants to rely on version `7.2`, which no longer needs the package `py` to be installed (see note [here](https://pylib.readthedocs.io/en/stable/)).

### PR DESCRIPTION
**Pull request by krzywon**
This is a general cleanup of the CI.

- Added `requirements.txt`
- Removed testing dependencies from `setup.py`
- Unpinned dependency versions
- Incremented python versions

Fixes #32

### CODE REVIEW

**Comment by butlerpd** [comment]
@krzywon and @lucas-wilkins discuss where to put it.

**Comment by lucas-wilkins** [comment]
I think requirements.txt should be in the root directory, unless you have any objections @krzywon

**Comment by krzywon** [comment]
> I think requirements.txt should be in the root directory, unless you have any objections @krzywon

No objections. Making the change now.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Data import/export requirements
+h5py
+lxml
+
+# Calculation
+numpy
+
+# Unit testing
+pytest
+unittest-xml-reporting
```
[the remark this answers is not in the document]
Comment (krzywon) [reply requirements.txt:10]: Like I said in the meeting, `xmlrunner` and `unittest-xml-reporting` are the same thing: https://github.com/xmlrunner/unittest-xml-reporting

**Comment by butlerpd** [comment]
@lucas-wilkins are you happy with this? If so can you submit an "approve"

**Review by lucas-wilkins — APPROVED** [review]

**Pull request merged**